### PR TITLE
fix: deprecate use of downshift

### PR DESCRIPTION
**Pull request by vacas5**
In an attempted upgrade of downshift I noticed a bug in their types.
Looking into a possible contribution, I then noticed that they have not
merged a PR in about a year. I am not sure if the project is currently
maintained, as all PRs were failing CI.

Instead I am attempting to remove downshift as a dependency altogether,
which should give users more granular control of accessibility concerns.

BREAKING CHANGE:
I will no longer return all of the Downshift props from the onSelect
callback in DropdownMenu.

## Checklist for PR creator

- [ ] If any new components were added, there are exported from `packages/index.ts`
- [ ] If this PR is associated with a JIRA, it is mentioned in commit message footer ("Closes …")
- [ ] If this PR contains breaking changes, is stated in commit message body ("BREAKING CHANGE: …")
- [ ] Info for applicable sections above is provided

### CODE REVIEW

**Review by nataliepina — CHANGES_REQUESTED** [review]
Screen Shot 2022-05-17 at 8 39 35 AM
Comparing the changes and it looks like there are some missing flex props that should be passed down to a `div` that is a child of `PopoverListItem` in the `DropdownMenu`. These look to have come from along with Downshift. 

Missing flex props can be seen here:
Screen Shot 2022-05-17 at 8 58 11 AM

File: tsconfig.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "moduleResolution": "node",
     "skipLibCheck": true,
     "tsBuildInfoFile": ".tsbuildinfo",
-    "lib": ["es6", "dom", "es2017"]
```
Comment (seialkali) [inline tsconfig.json:20]: Nice. Just out of curiosity, did adding this fix any issues we were having with TS?

File: packages/popover/components/PopoverListItemIcon.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ interface PopoverListItemProps extends IconProps {
   position?: "start" | "end";
 }
 
-const PopoverListItemIcon: React.SFC<PopoverListItemProps> = props => (
+const PopoverListItemIcon = (props: PopoverListItemProps) => (
```
Comment (nataliepina) [inline packages/popover/components/PopoverListItemIcon.tsx:11]: 👏

**Comment by vacas5** [comment]
I will more than likely close this, but I'll try and grab some of the nicer parts into another PR

**Comment by vacas5** [comment]
For now, I've gotten the upgrade to work. We can possibly revisit this if we find that downshift is indeed abandoned

**Pull request closed without merging**